### PR TITLE
[Feat] [snt9|2_5] Added a case: Resnet50-based  Vehicle identification

### PR DESCRIPTION
**Pull request by ilovexsir**
## Test environment
Hardware: 1*ascend-snt9|ARM: 24核 96GB
Image: python3.9-ms2.5.0-cann8.0.0.beta1
## Test Report

Accuracy: 60.1%
Inference Time: 0.07 ms per image
Training Time: 50 minutes
Total Script Runtime: 55 minutes

## Related Issues

Fixes #4

### CODE REVIEW

**Comment by hongziqi** [comment]
`dev/ascend/snt9/2_5/基于Mindspore实现贫血症识别.ipynb` had merged. You should synchronize the main repository first. 
Also `dev/ascend/snt9/2_5/基于Mindspore实现肺炎识别.ipynb` had submitted in PR #7.

**Comment by ilovexsir** [comment]
I update it, please review it.

File: dev/ascend/snt9/2_5/基于Mindspore实现车辆分类.ipynb [hunk 1]
```diff
@@ -0,0 +1,920 @@
...
+    "\n",
+    "for epoch in range(num_epochs):\n",
+    "    curr_loss = train(data_loader_train, epoch)\n",
+    "    curr_acc = evaluate(data_loader_val)\n",
+    "    if (epoch + 1) % 10 == 0:\n",
+    "        print(\"-\" * 50)\n",
+    "        print(\"Epoch: [%3d/%3d], Average Train Loss: [%5.3f], Accuracy: [%5.3f]\" % (\n",
+    "            epoch+1, num_epochs, curr_loss, curr_acc\n",
+    "        ))\n",
+    "        print(\"-\" * 50)\n",
```
Comment (hongziqi) [inline dev/ascend/snt9/2_5/基于Mindspore实现车辆分类.ipynb:811]:
```suggestion
```

File: dev/ascend/snt9/2_5/基于Mindspore实现车辆分类.ipynb [hunk 2]
```diff
@@ -0,0 +1,920 @@
...
+     },
+     "execution_count": 2,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "from download import download\n",
+    "\n",
+    "url = \"https://ai-studio-online.bj.bcebos.com/v1/c82c32c336214e138b7839ed1aeb0e5ec292ea76d7da4ada98f9d590370e4cd1?responseContentDisposition=attachment%3Bfilename%3Dcar.zip&authorization=bce-auth-v1%2F5cfe9a5e1454405eb2a975c43eace6ec%2F2025-03-09T04%3A15%3A52Z%2F21600%2F%2F3dc91d399aa3c4be7e041611baedd2bd1b2f57dcec21fb55a2dfbbc3e5a35f47\"\n",
```
Comment (hongziqi) [inline dev/ascend/snt9/2_5/基于Mindspore实现车辆分类.ipynb:124]: To ensure broader accessibility and avoid authentication-related issues with dataset links, we recommend using these open dataset platforms instead, you can push the dataset in these platform:
* [modelers.cn](https://modelers.cn/datasets)
* [xihe.mindspore.cn](https://xihe.mindspore.cn/datasets)

Comment (ilovexsir) [reply dev/ascend/snt9/2_5/基于Mindspore实现车辆分类.ipynb]: OK, I will update it.

**Comment by ilovexsir** [comment]
I already update it, please review it.

File: dev/ascend/snt9/2_5/ResNet50实现车辆识别.ipynb [hunk 3]
```diff
@@ -0,0 +1,909 @@
...
+       "'./dataset'"
+      ]
+     },
+     "execution_count": 2,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "from download import download\n",
```
Comment (hongziqi) [inline dev/ascend/snt9/2_5/ResNet50实现车辆识别.ipynb:122]:
```suggestion
    "!git clone https://modelers.cn/xsir/Vehicle-identification.git\n",
    "!unzip -q Vehicle-identification/dataset.zip -d Vehicle-identification/"
```
Please use official download method

File: dev/ascend/snt9/2_5/ResNet50实现车辆识别.ipynb [hunk 4]
```diff
@@ -0,0 +1,909 @@
...
+    "import mindspore as ms\n",
+    "import mindspore.dataset as ds\n",
+    "import mindspore.dataset.vision as vision\n",
+    "from typing import Type, Union, List, Optional\n",
+    "from mindspore import nn, train\n",
+    "from mindspore.common.initializer import Normal\n",
+    "import os\n",
+    "import time\n",
+    "# 数据集目录路径\n",
+    "data_path_train = \"./dataset/train/\"\n",
```
Comment (hongziqi) [inline dev/ascend/snt9/2_5/ResNet50实现车辆识别.ipynb:162]:
```suggestion
    "data_path_train = \"./Vehicle-identification/train/\"\n",
```

File: dev/ascend/snt9/2_5/ResNet50实现车辆识别.ipynb [hunk 5]
```diff
@@ -0,0 +1,909 @@
...
+    "data_path_val = \"./dataset/val/\"\n",
```
Comment (hongziqi) [inline dev/ascend/snt9/2_5/ResNet50实现车辆识别.ipynb:163]:
```suggestion
    "./Vehicle-identification/val/\"\n",
```

Comment (ilovexsir) [reply dev/ascend/snt9/2_5/ResNet50实现车辆识别.ipynb:163]: I try download code, which can normal run.

File: dev/ascend/snt9/2_5/ResNet50实现车辆识别.ipynb (hunk 3 above)
Comment (ilovexsir) [reply dev/ascend/snt9/2_5/ResNet50实现车辆识别.ipynb:122]: I try it, but the environment can't use git-lfs.

Comment (hongziqi) [reply dev/ascend/snt9/2_5/ResNet50实现车辆识别.ipynb:122]: The environmemt had installed git-lfs (check if git-lfs is installed by `git lfs install` on the terminal) , you can comment here on the error you encountered.

Comment (hongziqi) [reply dev/ascend/snt9/2_5/ResNet50实现车辆识别.ipynb:122]: What image did you choose? git-lfs is only supported in `python3.9-ms2.3.0-cann8.0.RC2.beta1` and above.
(`python3.9-ms2.2.14-cann7.0.0.beta1` is not supported)

Comment (ilovexsir) [reply dev/ascend/snt9/2_5/ResNet50实现车辆识别.ipynb]: OK, i choose a wrong image. I already update it.

**Comment by hongziqi** [comment]
Please squash multiple commits into a single commit

**Comment by ilovexsir** [comment]
I update it.

**Review by hongziqi — APPROVED** [review]
